### PR TITLE
[CIS-1525] Crashfix for hanging DispatchWorkItem reference in WebSocketClient

### PR DESCRIPTION
**Pull request by bielikb**
### 🔗 Issue Link
CIS-1525

### 🎯 Goal

_Describe why we are making this change._

### 🛠 Implementation

_Provide a description of the implementation._

### 🎨 Changes

* `WebSocketPingController` is run on specific engineQueue
* URL socket client calls delegate methods on dedicated `callbackQueue`
* Few renamings & doc updates

### 🧪 Testing

_Describe the steps how this change can be tested (or why it can't be tested)._

### ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChat/Utils/Timers.swift [hunk 1]
```diff
@@ -84,17 +84,17 @@ private class RepeatingTimer: RepeatingTimerControl {
...
-        timer.setEventHandler(handler: onFire)
+        timer?.schedule(deadline: .now() + .milliseconds(Int(timeInterval)), repeating: timeInterval, leeway: .seconds(1))
+        timer?.setEventHandler(handler: onFire)
     }
     
     deinit {
-        timer.setEventHandler {}
-        timer.cancel()
+        timer?.setEventHandler {}
+        timer?.cancel()
```
Comment (nuno-vieira) [inline Sources/StreamChat/Utils/Timers.swift:97]:
```suggestion
        timer?.setEventHandler {}
        time?.invalidate()
        timer?.cancel()
```
Don't we need to call invalidate here? I don't remember if this is still necessary or not 🤔

Comment (bielikb) [reply Sources/StreamChat/Utils/Timers.swift]: We're leveraging `DispatchSourceTimer` internally.

File: Sources/StreamChat/WebSocketClient/WebSocketPingController.swift [hunk 2]
```diff
@@ -6,15 +6,15 @@ import Foundation
 
 /// A delegate to control `WebSocketClient` connection by `WebSocketPingController`.
 protocol WebSocketPingControllerDelegate: AnyObject {
-    /// `WebSocketPingController` will call it method periodically to keep a connection alive.
+    /// `WebSocketPingController` will call this func periodically to keep a connection alive.
     func sendPing()
     
-    /// `WebSocketPingController` will call it to force disconnect `WebSocketClient`.
+    /// `WebSocketPingController` will call this func to force disconnect `WebSocketClient`.
```
Comment (polqf) [inline Sources/StreamChat/WebSocketClient/WebSocketPingController.swift:12]: I would vote for either function or method, but not func

File: Sources/StreamChat/WebSocketClient/WebSocketPingController.swift [hunk 3]
```diff
@@ -50,33 +51,48 @@ class WebSocketPingController {
...
     
-    func pongRecieved() {
+    func pongReceived() {
         log.info("WebSocket Pong")
         cancelPongTimeoutTimer()
     }
     
+    // MARK: Timers
+    
+    private func createPingTimerIfNeeded() {
```
Comment (polqf) [inline Sources/StreamChat/WebSocketClient/WebSocketPingController.swift:80]: `schedulePingTimerIfNeeded`?

**Review by polqf — APPROVED** [review]
Just a couple of wording comments

File: Sources/StreamChat/WebSocketClient/WebSocketPingController.swift (hunk 2 above)
Comment (bielikb) [reply Sources/StreamChat/WebSocketClient/WebSocketPingController.swift:12]: Yeah sure ;)

Comment (bielikb) [reply Sources/StreamChat/WebSocketClient/WebSocketPingController.swift]: Done!

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**